### DISCUSSION TITLE
PML in Plot2D

### DISCUSSION THREAD

**Discussion by shawnszhao** [NanoComp/meep#2147] [Q&A] [answered]
I want to be sure I am understanding the PML behavior correctly. I am performing a 3D simulation and visualizing a 2D plane using Plot2D. However, the PML is drawn over the entire plane. Following from https://github.com/NanoComp/meep/issues/1046#issuecomment-550118126, the output shown there looks like:
![image](https://user-images.githubusercontent.com/13872547/180487042-3d8dbe21-c820-4731-9195-84c7a7f3b7dd.png)

If I run the same thing on my machine, I get hatches throughout the entire plane. 
![output](https://user-images.githubusercontent.com/13872547/180488536-9480d3e4-4e0f-431f-b058-4a8938dcfafb.png)

If I plot: 
`f = plt.figure()
output_plane = mp.Volume(center=mp.Vector3(y=2),size=mp.Vector3(10,0,10))
mp.visualization.plot_boundaries(sim,ax=f.gca(),output_plane=output_plane)` 
![output](https://user-images.githubusercontent.com/13872547/180489040-e8451861-1f02-496e-afd7-0b0985b2bbae.png)
I notice that the axis goes from 0-1. Not sure if that is relevant. There does not appear to be attenuation throughout the entire 3D volume so I do not think there is _actually_ a PML everywhere, but rather just an issue with the plotting. However, I would like to be sure. 

Thanks!
Shawn

**Comment by oskooi** [comment, marked as answer]
#2079 fixed a bug in `plot2D` and PML in a 3D cell. You can check whether this fixes your problem by using the Conda package (which was updated to [version 1.24](https://github.com/NanoComp/meep/releases/tag/v1.24.0) released yesterday) or build from source using the master branch.

**Reply by shawnszhao** [reply]
Thanks for the quick response. New version works well (with nice-looking plane waves I might add).
